Assets/scripts/BattleManager.cs

```diff
@@ -117,6 +117,8 @@ private class PlayerBattleData {
 		// Prepare battle data per player
 		attacker = players [m.AttackingUnit.PlayerNumber];
 		defender = players [m.DefendingUnit.PlayerNumber];
+		attacker.unit = m.AttackingUnit;
+		defender.unit = m.DefendingUnit;
```

Comment:
These two lines are all the changes needed. You can revert all other changes in this file.